source/Calamari/Deployment/PackageRetention/PackageJournalCommandDecorator.cs

```diff
@@ -26,7 +28,21 @@ public PackageJournalCommandDecorator(ILog log, ICommandWithArgs command, IManag
 
         public int Execute(string[] commandLineArguments)
         {
-            journal.RegisterPackageUse();
+            // ReSharper disable once InvertIf
+            if (variables.IsPackageRetentionEnabled())
```

Comment:
Is this check needed if the journal does the same check in `RegisterPackageUse`?

Reply:
Good question.  I added this because the PackageIdentity.CreatePackageIdentity check can be fairly involved - it can potentially hit the file system, the journal, and the command-line args.  This way we don't have to mess around with all those things if we're not actually using the results.